runtime/mbarrier.cu

```diff
@@ -35,6 +35,16 @@ __device__ inline uint64_t arrive(uint32_t smem_barrier_ptr) {
   return state;
 }
 
+__device__ inline uint64_t arriveExpectTX(
+    uint32_t smem_barrier_ptr,
+    uint32_t tx_count) {
+  volatile uint64_t state;
+  asm volatile("mbarrier.arrive.expect_tx.shared.b64 %0, [%1], %2;\n"
```

Comment:
Do we need a protect? [doc](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-mbarrier-expect-tx-operation) says `Qualifier .expect_tx requires sm_90 or higher.`

Reply:
Good catch! Guard added.